src/sparseml/pytorch/torchvision/train.py

```diff
@@ -568,9 +568,17 @@ def log_metrics(tag: str, metrics: utils.MetricLogger, epoch: int, epoch_step: i
             loggers=logger,
             distillation_teacher=args.distill_teacher,
         )
-        optimizer = manager.modify(
-            model, optimizer, steps_per_epoch=steps_per_epoch, epoch=args.start_epoch
+        step_wrapper = manager.modify(
```

Comment:
@corey-nm We should pass the loggers into here so that from now on logging could be handled both by the modifiers (relevant for e.g. logging different quantization losses) and by the loss update of the wrapped manager (i.e. the RecipeManagerStepWrapper). It also prepares us well for distillation in the future.

I think we also need to recompute the steps per epoch passed in, taking into account grad accum steps and multi gpus.  Look at this line and those right above: https://github.com/neuralmagic/sparseml/blob/main/src/sparseml/transformers/sparsification/trainer.py#L272

After that, we can remove all the logging from this script (except perhaps for the python logger).

Reply:
Btw, looks like we can unify scaler and optimizer into "step_wrapper" and just pass it into the `train_one_epoch`(instead of having 2 separate/mutually exclusive objects)?

Reply:
1. the loggers were passed into manager.initializer() in the call right above this - does that not pass the loggers to all the modifiers?
2. steps_per_epoch does take into account gradient accumulation, but not multi gpu. i can fix the multi gpu in a separate pr
3. yeah the scaler/optimizer interactions are weird. you need to pass optimizer to scaler.step, and you also need to call optimizer.zero_grad() through the optimizer. it seems like if they were designed differently they could be combined. we could create a new class that wraps them, but i'm not sure the indirection it would add would be worth it. thoughts?

Reply:
I see, you're right about the loggers init. 
Thanks for taking on 2. For 3, no I didn't mean to push for it unless it's an easy clean up.

Reply:
I think let's push this in and continue to improve it